File: app/src/main/java/me/pedroguimaraes/weatherdemo/WeatherApplication.java
package me.pedroguimaraes.weatherdemo;

import android.app.Application;
import android.content.Context;
import com.squareup.leakcanary.LeakCanary;

public class WeatherApplication extends Application {

    private static WeatherApplication instance;

    public static Context getContext() {
        return instance.getApplicationContext();
    }

    @Override
    public void onCreate() {
        instance = this;
        super.onCreate();
        startLeakCanary();
    }

    private void startLeakCanary() {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return;
        }
        LeakCanary.install(this);
    }
}

File: app/src/main/java/me/pedroguimaraes/weatherdemo/WeatherApplication.kt
package me.pedroguimaraes.weatherdemo

import android.app.Application
import android.content.Context
import com.squareup.leakcanary.LeakCanary

class WeatherApplication : Application() {

    override fun onCreate() {
        instance = this
        super.onCreate()
        startLeakCanary()
    }

    private fun startLeakCanary() {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return
        }
        LeakCanary.install(this)
    }

    companion object {

        private lateinit var instance: WeatherApplication

        val context: Context
            get() = instance.applicationContext
    }
}
